pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -253,20 +309,47 @@ func (tc *testCase) verifyResults(t *testing.T) {
 	}
 }
 
-func (tc *testCase) runTest(t *testing.T) {
-	testClient := tc.prepareTestClient(t)
-	metricsClient := metrics.NewHeapsterMetricsClient(testClient, metrics.DefaultHeapsterNamespace, metrics.DefaultHeapsterScheme, metrics.DefaultHeapsterService, metrics.DefaultHeapsterPort)
-	hpaController := NewHorizontalController(testClient.Core(), testClient.Extensions(), testClient.Extensions(), metricsClient, 0)
-	stop := make(chan struct{})
-	defer close(stop)
-	go hpaController.Run(stop)
-	if tc.verifyEvents {
-		// We need to wait for events to be broadcasted (sleep for longer than record.sleepDuration).
-		time.Sleep(12 * time.Second)
+func runTest(t *testing.T, tc *testCase) {
```

Comment:
> I don't think making a copy is a great idea. The test case contains things a simple shallow copy doesn't clone.

I don't think it matters, we are not modifying those values.

> Could we tests as additive as possible for this PR (even if we duplicate a couple functions)? Want to make sure we don't regress anything at this point

Fair point. I am adding additional tests instead.